docs/pipeline.md

```diff
@@ -0,0 +1,52 @@
+# Pipeline
+
+A Fondant pipeline is a tool for building complex workflows by creating a Directed Acyclic Graph (DAG) of different components that need to be executed. With Fondant, you can use both reusable components and custom components to construct your pipeline. Fondant extends the functionality of Kubeflow and provides additional features to simplify pipeline development.
+
+
+## Building a pipleine 
+
+To build a pipeline, you need to define a set of component operations called `ComponentOp`. A component operation encapsulates the specifications of the component and its runtime configuration.
+
+The component specifications include the location of the Docker image in the artifact registry. It is important to ensure that your Kubeflow service account has access to the image registry if it is not public.
+
+The runtime configuration consists of the component's arguments and the definition of node pools and resources. For example, if a component requires GPU for model inference, you can specify the necessary GPU resources in the runtime configuration.
+
+Here is an example of how to build and compile a pipeline using Fondant:
+```python
+from fondant.pipeline import ComponentOp, Pipeline, Client
```

Comment:
will we document the client separately ? If not I would add a section in this file explaing the host and base_path

Reply:
I think it does not have a lot of functionalities but I would aim to have more documentation around it at some point. We can link to your infra docs since they also have explanation about how to fetch the `host_url`. The `base_path` is mentioned below. 

I will add a reference to your docs here